src/mmrelay/radio/message.py

```
@@ -0,0 +1,73 @@
+    location: dict[str, float] | None = None
+    telemetry: dict[str, Any] | None = None
+
+    def __post_init__(self) -> None:
+        """
+        Validate message fields after initialization.
+
+        Ensures critical fields are present and valid. Raises ValueError
+        for invalid messages.
+        """
+        if not self.text:
+            raise ValueError("RadioMessage.text cannot be empty")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)